ENVParser/Utils/CsvExporter.cs

```diff
@@ -0,0 +1,97 @@
+﻿using CsvHelper;
+using ENVParser.Fields;
+using System.Globalization;
+
+namespace ENVParser.Utils
+{
+    internal class CsvExporter
+    {
+        private readonly HashSet<string> _validFieldsForRGBValues;
+
+        // Defines the structure of CSV records
+        private class CsvOutput
+        {
+            public string FieldName { get; set; }
+            public string Value { get; set; }
+            public float? RGBValue { get; set; }
+            public string FieldType { get; set; }
+        }
+
+        public CsvExporter()
+        {
+            _validFieldsForRGBValues = RGBFieldNameProvider.GetValidFields();
+        }
+
+        public void Export(string filePath, EnvFile envFile)
+        {
+            if (string.IsNullOrEmpty(filePath))
+            {
+                throw new ArgumentNullException(nameof(filePath), "An output file path must be supplied.");
+            }
+            string outputDirectory = Path.GetDirectoryName(filePath) ?? throw new InvalidOperationException("Failed to get directory name from file path.");
+
+            // As this is a one-way transfer
+            // No need for special override for Texture Section so, can override value to 0
+            var csvData = envFile.Select(data => new CsvOutput
+            {
+                FieldName = data.Key,
+                Value = !data.Key.Equals("UnusedTextureSection") ? data.Value.ToString() : "0",
+                RGBValue = _validFieldsForRGBValues.Contains(data.Key) ? GetRGBValue(data.Value) : null,
+                FieldType = TextReadableFieldType(data.Value.GetType().Name),
+            }
+            ).ToList();
+
+
+
+            using var writer = new StreamWriter(filePath);
+            using var csv = new CsvWriter(writer, CultureInfo.InvariantCulture);
+            Console.WriteLine("Writing csv...");
+            csv.WriteRecords(csvData);
+        }
+
+        private static string TextReadableFieldType(string fieldType)
+        {
+
+            string readableFieldType;
+            switch (fieldType)
+            {
```

Comment:
This definitely looks similar to some stuff in the JSON code. Is this sort of thing what you mean when you mention refactoring later to consolidate pieces of the code?

Reply:
Yes, along with the actual way I've implemented the CSV export.